workflow/rocoto/gefs_tasks.py

```diff
@@ -9,48 +10,63 @@ def __init__(self, app_config: AppConfig, cdump: str) -> None:
         super().__init__(app_config, cdump)
 
     def stage_ic(self):
-
         cpl_ic = self._configs['stage_ic']
-
         deps = []
-
+        if self._configs['base']['END_OF_IAU_START']:
+            DTG = self._configs['base']['SDATE'] + timedelta(hours=3)
+            DTG_PREFIX = DTG.strftime('%Y%m%d') + '.' + DTG.strftime('%H') + '0000'
+        else:
+            DTG_PREFIX = "@Y@m@d.@H0000"
         # Atm ICs
         if self.app_config.do_atm:
             prefix = f"{cpl_ic['BASE_CPLIC']}/{cpl_ic['CPL_ATMIC']}/@Y@m@d@H/mem000/atmos"
-            for file in ['gfs_ctrl.nc'] + \
-                        [f'{datatype}_data.tile{tile}.nc'
-                         for datatype in ['gfs', 'sfc']
-                         for tile in range(1, self.n_tiles + 1)]:
-                data = f"{prefix}/{file}"
+            if self._base['EXP_WARM_START']:
+                for file in ['fv_core.res.nc'] + \
+                            [f'{datatype}.tile{tile}.nc'
+                             for datatype in ['ca_data', 'fv_core.res', 'fv_srf_wnd.res', 'fv_tracer.res', 'phy_data', 'sfc_data']
+                             for tile in range(1, self.n_tiles + 1)]:
+                    data = f"{prefix}/{DTG_PREFIX}.{file}"
+                    dep_dict = {'type': 'data', 'data': data}
+                    deps.append(rocoto.add_dependency(dep_dict))
+                prefix = f"{cpl_ic['BASE_CPLIC']}/{cpl_ic['CPL_ATMIC']}/@Y@m@d@H/mem000/med"
+                data = f"{prefix}/{DTG_PREFIX}.ufs.cpld.cpl.r.nc"
                 dep_dict = {'type': 'data', 'data': data}
```

Comment:
The `add_dependency()` method can already handle these complicated offset situations (we need to add an offset to the filename but not the directory). We just need to pass an array of data and offsets and they will be combined:
```suggestion
                    data = [prefix, f"{dtg_prefix}.{file}"]
                    dep_dict = {'type': 'data', 'data': data, 'offset': [None, offset]}
                    deps.append(rocoto.add_dependency(dep_dict))
                prefix = f"{cpl_ic['BASE_CPLIC']}/{cpl_ic['CPL_ATMIC']}/@Y@m@d@H/mem000/med"
                data = [prefix, f"{dtg_prefix}.ufs.cpld.cpl.r.nc"]
                dep_dict = {'type': 'data', 'data': data, 'offset': [None, offset]}
```